api/users/views.py

```diff
@@ -425,12 +425,85 @@ def get_default_odm_query(self):
             Q('is_registration', 'eq', True) &
             Q('contributors', 'eq', user._id)
         )
+
         permission_query = Q('is_public', 'eq', True)
         if not current_user.is_anonymous():
             permission_query = (permission_query | Q('contributors', 'eq', current_user._id))
         query = query & permission_query
         return query
 
+class UserLogs(JSONAPIBaseView, generics.ListCreateAPIView, ODMFilterMixin, UserMixin):
+    """ Logs from nodes a user is currently associated with.
+    This returns all logs from every node the user is currently associated with. This means
+    this might return logs from other users, and might not return all logs associated with
+    this particular user (if that user is no longer contributing to a node).
+
+    ##Attributes
+    <!--- Copied Attributes from LogList -->
+
+    OSF Log entities have the "logs" `type`.
+
+        name           type                   description
+        ----------------------------------------------------------------------------
+        date           iso8601 timestamp      timestamp of Log creation
+        action         string                 Log action
+
+    ##Query Params
+    <!--- Copied Query Params from LogList -->
+
+    Logs may be filtered by their `action` and `date`.
+
+    Also, this can take in `aggregates=true` as a param, which then will include in the meta
+    field a count of how many of each type of 4 types of actions would be returned (regardless
+    of pagination). Those types are files, comments, wiki, and nodes, each associated with a
+    subset of log actions. It will also return the last loggable action within the logs returned.
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        CurrentUser,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.USERS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = NodeLogSerializer
+    pagination_class = UserLogPagination
+    view_category = 'users'
+    view_name = 'user-logs'
+
+    ordering = ('-date', )
+
+    def get_default_odm_query(self):
+        user = self.get_user()
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        raw_nodes = list(Node.find(
+            Q('contributors', 'eq', user) &
+            Q('is_folder', 'ne', True) &
+            Q('is_deleted', 'ne', True)
+        ))
+        nodes = [each for each in raw_nodes if each.is_public or each.can_view(auth)]
+        if not nodes:
+            raise NotFound
+        query = [Q('params.node', 'eq', node._id) for node in nodes]
+        return reduce(lambda x, y: x | y, query)
+
+    def get_queryset(self):
+        self.paginator.node_log_aggregates = self.get_aggregates
```

Comment:
`get_aggregates` calls `get_query_from_request`, then `get_query_from_request` gets call in the line after this, which seems inefficient, esp. because there's a lot of querying and filtering going on in `get_default_odm_query`. 

Can you refactor this to save a call to `get_query_from_request`? Maybe use `functools.partial`:


```python
def get_aggregates(self, query):
    #...

def get_queryset(self):
    query = self.get_queryset_from_request()
    self.paginator.node_log_aggregates = partial(self.get_aggregates, query)
```